tools/xtce_generation/xtce_generator_template.py

```diff
@@ -1,47 +1,104 @@
-"""
-Module for generating XTCE files for telemetry packets for a given instrument.
-
-This module provides functionality to generate XTCE files for telemetry packets
-for an instrument. It includes a `TelemetryGenerator` class for creating XTCE files
-based on packet definitions stored in an Excel file.
-"""
+"""XTCE generator script that uses CLI for generalized format."""
 
+import argparse
+import json
 from pathlib import Path
 
+import imap_data_access
+
 from tools.xtce_generation.telemetry_generator import TelemetryGenerator
 
 
+# Function to parse command line arguments
+def _parse_args():
+    """
+    Parse the command line arguments.
+
+    The expected input format is:
+    --instrument "instrument_name"
+    --file_path "full_file_path"
+    --packets '{"packet_name_1": app_id_1, "packet_name_2": app_id_2}'
+
+    Returns
+    -------
+    args : argparse.Namespace
+        An object containing the parsed arguments and their values.
+    """
+    description = (
+        "This command line program generates an instrument specific XTCE file."
+        "Example usage: "
+        '--instrument "swapi"'
+        "--file_path "
+        '"/Users/anma6676/Desktop/Repositories/imap_processing/tools/xtce_generation/TLM_SWP_20231006-121021.xlsx"'
+        "--packets '"
+        '{"P_SWP_HK": 1184, '
+        '"P_SWP_SCI": 1188, '
+        '"P_SWP_AUT": 1192, '
+        "}'"
+    )
+    instrument_help = (
+        "The instrument to process. Acceptable values are: "
+        f"{imap_data_access.VALID_INSTRUMENTS}"
+    )
+    file_path_help = "Provide full file path to write packets to."
+    packets_help = (
+        "Provide packet dictionary using packet_name, and app_id."
+        '{"<packet_name>": <app_id>}'
+    )
+
+    parser = argparse.ArgumentParser(prog="imap_xtce", description=description)
+    parser.add_argument("--instrument", type=str, required=True, help=instrument_help)
+    parser.add_argument("--file_path", type=str, required=True, help=file_path_help)
```

Comment:
It should be hyphens for the CLI input to be consistent with our other CLIs. That then gets transformed to underscores within the Python code for you.
```suggestion
    parser.add_argument("--file-path", type=str, required=True, help=file_path_help)
```